### PR TITLE
Compatible with Go v1.22+

### PR DESCRIPTION
**Pull request by attiasas**
- [x] All [tests](https://github.com/jfrog/gomod-absolutizer#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

Adding support to Go `v1.22+`
This change raises the minimum Go version to `v1.18`

### CODE REVIEW

File: .github/workflows/analysis.yml [hunk 1]
```diff
@@ -22,11 +10,12 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@v3
         with:
-          go-version: 1.17.x
+          go-version: 1.20.x
```
Comment (yahavi) [inline .github/workflows/analysis.yml]: Why not 18?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest, macOS-latest]
-        go: ["1.16", "1.17", "1.18", "1.19"]
+        go: ["1.18", "1.19", "1.22.3"]
```
Comment (yahavi) [inline .github/workflows/test.yml:12]: Wouldn't that work?
```suggestion
        go: ["1.18", "1.19", "1.22"]
```

File: testdata/version/minor/go.mod [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module test-module
+go 1.12
```
Comment (yahavi) [inline testdata/version/minor/go.mod]:
```suggestion
go 1.18
```

File: .github/workflows/analysis.yml [hunk 4]
```diff
@@ -22,11 +10,12 @@ jobs:
...
       - name: Static Code Analysis
-        uses: dominikh/staticcheck-action@v1
+        uses: golangci/golangci-lint-action@v3
         with:
-          install-go: false
+          args: |
+            --timeout 5m --out-${NO_FUTURE}format colored-line-number --enable errcheck,gosimple,govet,ineffassign,staticcheck,typecheck,unused,gocritic,asasalint,asciicheck,errchkjson,exportloopref,forcetypeassert,makezero,nilerr,unparam,unconvert,wastedassign,usestdlibvars
   Go-Sec:
```
Comment (yahavi) [inline .github/workflows/analysis.yml]: You can simply add "gosec" to the golangci-lint-action

**Review by yahavi — CHANGES_REQUESTED** [review]

**Comment by attiasas** [comment]
duplicate with: https://github.com/jfrog/gomod-absolutizer/pull/6

**Pull request closed without merging**